core/src/main/java/org/elasticsearch/cluster/routing/allocation/AllocationService.java

```diff
@@ -143,7 +176,7 @@ public AllocationService(Settings settings, AllocationDeciders allocationDecider
      * If the same instance of the routing table is returned, then no change has been made.
      */
     public RoutingAllocation.Result reroute(ClusterState clusterState) {
-        return reroute(clusterState, false);
+        return reroute(clusterState, "reroute");
```

Comment:
I agree w.r.t. production code. This method is just called from tests and I did not want to touch dozens of classes just for that. But you're right, this stops nobody from calling this method from production code so it's safer to just remove this method and adapt the tests.